proto/lbm/collection/v1/collection.proto

```diff
@@ -0,0 +1,186 @@
+  string meta = 3;
+  // decimals is the number of decimals which one must divide the amount by to get its user representation.
+  int32 decimals = 4;
+  // mintable represents whether the token class is allowed to mint or burn its tokens.
+  bool mintable = 5;
+}
+
+// NFTClass defines the class of non-fungible token.
+//
+// Since: 0.46.0 (finschia)
+message NFTClass {
+  option (gogoproto.goproto_getters)         = true;
+  option (cosmos_proto.implements_interface) = "TokenClass";
+
+  // id defines the unique identifier of the token class.
+  // Note: size of the class id is 8 in length.
+  string id = 1;
+  // name defines the human-readable name of the token class.
+  string name = 2;
+  // meta is a brief description of the token class.
+  string meta = 3;
+}
+
+// NFT defines the information of non-fungible token.
+//
+// Since: 0.46.0 (finschia)
+message NFT {
+  // id defines the unique identifier of the token.
+  string id = 1;
+  // name defines the human-readable name of the token.
+  string name = 2;
+  // meta is a brief description of the token.
+  string meta = 3;
+}
+
+// OwnerNFT defines the information of non-fungible token.
+//
+// Note: deprecated (use NFT)
+message OwnerNFT {
+  option (cosmos_proto.implements_interface) = "Token";
+
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // id defines the unique identifier of the token.
+  string token_id = 2;
+  // name defines the human-readable name of the token.
+  string name = 3;
+  // meta is a brief description of the token.
+  string meta = 4;
+
+  // owner of the token.
+  string owner = 5;
+}
+
+// FT defines the information of fungible token.
+//
+// Note: deprecated (use FTClass)
+message FT {
+  option (cosmos_proto.implements_interface) = "Token";
+
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // token id defines the unique identifier of the fungible token.
+  string token_id = 2;
+  // name defines the human-readable name of the fungible token.
+  string name = 3;
+  // meta is a brief description of the fungible token.
+  string meta = 4;
+  // decimals is the number of decimals which one must divide the amount by to get its user representation.
+  int32 decimals = 5;
+  // mintable represents whether the fungible token is allowed to be minted or burnt.
+  bool mintable = 6;
+}
+
+// TokenType defines the information of token type.
+//
+// Note: deprecated (use TokenClass)
+message TokenType {
+  // contract id associated with the contract.
+  string contract_id = 1;
+  // token type defines the unique identifier of the token type.
+  string token_type = 2;
+  // name defines the human-readable name of the token type.
+  string name = 3;
+  // meta is a brief description of the token type.
+  string meta = 4;
+}
+
+// Coin defines a token with a token id and an amount.
+message Coin {
+  option (gogoproto.equal) = true;
+
+  // token id associated with the token.
+  string token_id = 1;
```

Comment:
`token_id` means `denomination`?

Reply:
No, there is no denomination for the tokens in x/token. Daphne uses `token_id` directly.

Reply:
Sorry, I missunderstood that `Coin` is COIN.